Decision records/06 - We split up large component test files by feature.md

```diff
@@ -0,0 +1,41 @@
+# We split up large component test files by feature
+
+## Date
+
+July 30, 2018
+
+## Contributors
+
+* Chris Sauvé
+* Mallory Allen
+* Tzvi Melamed
+* Ismail Syed
+* Gord Pearson
+* Matt Seccafien
+
+## Summary
+
+When a test file for a component grows too large, it should split into several sibling test files, each covering a feature of the larger component.
+
+## History
+
+In Shopify Web, some of our components have grown extremely large, despite good separation of discrete logical areas into separate subcomponents. These components are usually those that represent a "page", and are often needed because of the need to collocate form data and/ or GraphQL handling.
+
+At the time of this decision, the largest of these components had test files anywhere from 3000–4500 lines long. Based on the complexity of other pages in the admin, we anticipate these not to be the largest test files in the app. The size of these files causes tooling like VSCode’s auto-formatting/ linting highlights not to function correctly, make navigating the file very difficult, and usually require a developer to focus a part of the test file to work on tests iteratively.
+
+## Decision
+
+Developer experience is a higher priority than technical purity, so we have decided these test files need to be split. Our existing directory structure can handle this fairly gracefully; a `Component/tests` directory already contains a single test file for the component (alongside directories for things like fixtures), so collocating all of the split test files in this directory is the natural next step.
```

Comment:
This also effects velocity (since stuff like prettier can be time consuming with auto format working properly), might want to call that out here.